packages/dd-trace/src/startup-log.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+
+const mainLogger = require('./log')
+const path = require('path')
+const { inspect } = require('util')
+const tracerVersion = require('../lib/version')
+
+let os
+try {
+  os = require('os')
+} catch (e) {
+  // in browser
+}
+
+const logger = Object.create(mainLogger)
+logger._enabled = process.env.DD_TRACE_STARTUP_LOGS !== '0'
+
+let config
+let plugins = []
+let samplingRules = []
+
+let alreadyRan = false
+
+function getIntegrations () {
+  const integrations = []
+  for (const plugin of plugins) {
+    if (plugin.versions) {
+      try {
+        const version = require(path.join(plugin.name, 'package.json')).version
+        integrations.push(`${plugin.name}@${version}`)
+      } catch (e) {
+        integrations.push(plugin.name)
+      }
+    } else {
+      integrations.push(plugin.name)
+    }
+  }
+  return Array.from(new Set(integrations))
+}
+
+function startupLog (agentError) {
+  if (!config) {
+    return
+  }
+
+  if (alreadyRan) {
+    return
+  }
+
+  alreadyRan = true
```

Comment:
How can it run multiple times?

Reply:
This function is called in the `platform.request` callback, so it's called on each HTTP request to the agent, but we only actually want it to run once (the first time).

Very open to any other approaches if you have ideas.

Reply:
Ah I see, that makes sense. Replacing the function with a no-op would be more efficient if no code in this file is ever going to run again, but if that's too much work it's fine as it is.

Reply:
Replacing the function requires accessing it from the exports each time, and I didn't want it to be that brittle.

Reply:
Not necessarily if there is a wrapper, but then that's arguably no better. I didn't understand at first that this was called on agent requests as well, so it makes a lot more sense now.